packages/itwin/property-grid/src/components/MultiElementPropertyGrid.tsx

```diff
@@ -63,35 +74,168 @@ export const MultiElementPropertyGrid = (props: PropertyGridProps) => {
             });
           }
         );
+
+        // if selection is not from us, clear the ancestry
+        if (undefined === args || args.source !== ProprtyGridSelectionScope) {
+          setAncestorKeys([]);
+        }
+
         setInstanceKeys(instanceKeys);
-        setMoreThanOneElement(selectionSet.instanceKeysCount > 1);
         setSelectedInstanceKey(undefined);
       }
     };
 
+    const onFrontstageReady = () => {
+      onSelectionChange();
+    };
+
     // ensure this selection handling runs if component mounts after the selection event fires:
     onSelectionChange();
 
     const removePresentationListener = Presentation.selection.selectionChange.addListener(onSelectionChange);
     // if the frontstage changes and a selection set is already active we need to resync this widget's state with that selection
-    const removeFrontstageReadyListener = FrontstageManager.onFrontstageReadyEvent.addListener(onSelectionChange);
+    const removeFrontstageReadyListener = FrontstageManager.onFrontstageReadyEvent.addListener(onFrontstageReady);
     return () => {
       removePresentationListener();
       removeFrontstageReadyListener();
     };
   }, [iModelConnection]);
 
+  useEffect(() => {
+    // update the ancestry when the instanceKeys change
+    if (instanceKeys.length === 1) {
+      const newAncestors = [...ancestorKeys, ...instanceKeys];
+      setAncestorKeys(newAncestors);
```

Comment:
shouldn't ancestorKeys be a dep of this useEffect? to prevent an infinite loop it'd need to ensure its not resetting the same keys, but seems like you could find yourself using old ancestor keys here right?

Reply:
Added ancestorKeys as dependency, had to ensure the new instanceKeys is not contained in the ancestoryKeys array to avoid an infinite loop.  Thinking about reworking instanceKeys and ancestoryKeys using useReducer, but that's for another day.